### PR TITLE
Improve lang lib Module.md fie

### PR DESCRIPTION
**Pull request by pubudu91**
Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.array/src/main/ballerina/src/lang.array/Module.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
 ## Module Overview
 
-This module provides lang library list operations defined by the language specification 2019R2.
+This module provides lang library list operations defined by the language specification 2019R3. All the functions listed 
+below are supported by resizable arrays. Fixed-length arrays and tuples do not support any operations that violate the 
+properties of these types. e.g., pushing a value to a tuple.
```
Comment (MaryamZi) [inline langlib/lang.array/src/main/ballerina/src/lang.array/Module.md]:
```suggestion
properties of these types. e.g., pushing a value to a fixed-length array.
```

Not sure the tuple example is correct. Even if we can't now, this should be possible for a tuple with a compatible rest field right?

File: langlib/lang.array/src/main/ballerina/src/lang.array/Module.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
 ## Module Overview
 
-This module provides lang library list operations defined by the language specification 2019R2.
+This module provides lang library list operations defined by the language specification 2019R3. All the functions listed 
+below are supported by resizable arrays. Fixed-length arrays and tuples do not support any operations that violate the
```
Comment (MaryamZi) [inline langlib/lang.array/src/main/ballerina/src/lang.array/Module.md]:
```suggestion
in this module are supported by lists that do not have a fixed length. Fixed-length arrays and tuples do not support any operations that violate the 
```

File: langlib/lang.map/src/main/ballerina/src/lang.map/Module.md [hunk 3]
```diff
@@ -1,3 +1,5 @@
 ## Module Overview
 
-This module provides lang library map operations defined by the language specification 2019R2.
+This module provides lang library map operations defined by the language specification 2019R3. All the functions listed
+below are supported by the `map` type. Records do not support any functions which violate their respective inherent
```
Comment (MaryamZi) [inline langlib/lang.map/src/main/ballerina/src/lang.map/Module.md]:
```suggestion
in this module are supported by mapping types. Records do not support any functions that violate their inherent 
```

File: langlib/lang.map/src/main/ballerina/src/lang.map/Module.md [hunk 4]
```diff
@@ -1,3 +1,5 @@
 ## Module Overview
 
-This module provides lang library map operations defined by the language specification 2019R2.
+This module provides lang library map operations defined by the language specification 2019R3. All the functions listed
+below are supported by the `map` type. Records do not support any functions which violate their respective inherent 
+types. e.g., removing a required field of a record
```
Comment (MaryamZi) [inline langlib/lang.map/src/main/ballerina/src/lang.map/Module.md]:
```suggestion
types. e.g., removing a required field of a record.
```

File: langlib/lang.array/src/main/ballerina/src/lang.array/Module.md (hunk 1 above)
Comment (hasithaa) [reply langlib/lang.array/src/main/ballerina/src/lang.array/Module.md]: Yes, Since this is module MD, I think simply saying 2019R3 is enough for now. We can improve the doc later.

File: langlib/lang.array/src/main/ballerina/src/lang.array/Module.md (hunk 2 above)
Comment (pubudu91) [reply langlib/lang.array/src/main/ballerina/src/lang.array/Module.md]: Isn’t it too wordy?

File: langlib/lang.map/src/main/ballerina/src/lang.map/Module.md (hunk 3 above)
Comment (pubudu91) [reply langlib/lang.map/src/main/ballerina/src/lang.map/Module.md]: Can’t say mapping types since records only support a subset of the functions..

Comment (pubudu91) [reply langlib/lang.map/src/main/ballerina/src/lang.map/Module.md]: Can’t say mapping types since records only support a subset of the functions..

File: langlib/lang.array/src/main/ballerina/src/lang.array/Module.md (hunk 2 above)
Comment (MaryamZi) [reply langlib/lang.array/src/main/ballerina/src/lang.array/Module.md]: It's OK IMO, and it's better to stick to spec terms right?

File: langlib/lang.map/src/main/ballerina/src/lang.map/Module.md (hunk 3 above)
Comment (MaryamZi) [reply langlib/lang.map/src/main/ballerina/src/lang.map/Module.md]: But saying `map` types might also be misleading, right? It's like they are only supported with `map`-typed variables. Then we talk about records, and there's a disconnect.

How about,
```suggestion
in this module are supported by mapping types, as long as the function does not violate the inherent type.
```

Functions like remove can also fail on maps, if immutable.

File: langlib/lang.array/src/main/ballerina/src/lang.array/Module.md (hunk 1 above)
Comment (pubudu91) [reply langlib/lang.array/src/main/ballerina/src/lang.array/Module.md]: Wrote it for the current functionality we have.
